### PR TITLE
Add management commands to setup linked domains

### PR DESCRIPTION
**Pull request by sravfeyn**
## Summary

Current process to setup domain links involve running HQ code in a Django shell. This moves the code to run into management commands.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage
 NA

### QA Plan
NA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/linked_domain/README.md [hunk 1]
```diff
@@ -74,27 +74,18 @@ Project data like forms and cases would not be shared.
 
 ### On 'master domain':
 
-```
-DomainLink.link_domains('https://url.of.linked.hq/a/linked_domain_name/', 'master_domain_name')
-```
+Run `add_downstream_domain` management command on source HQ.
```
Comment (proteusvacuum) [inline corehq/apps/linked_domain/README.md:77]:
```suggestion
Run `add_downstream_domain` management command on source HQ.

    $ ./manage.py add_downstream_domain --url {https://url.of.linked.hq/a/linked_domain_name/} --domain {upstream_domain_name}

```

File: corehq/apps/linked_domain/management/commands/add_downstream_domain.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+class Command(BaseCommand):
+    """
+    Links a remote domain as a downstream domain to a given upstream domain on this HQ
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('-r', '--url', required=True,
+                            help="URL of downstream domain to link https://url.of.linked.hq/a/linked_domain_name/")
+        parser.add_argument('-d', '--domain', required=True,
```
Comment (proteusvacuum) [inline corehq/apps/linked_domain/management/commands/add_downstream_domain.py:14]: I would call this arg `upstream_domain` to be explicit

File: corehq/apps/linked_domain/management/commands/add_downstream_domain.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+from corehq.apps.linked_domain.models import DomainLink
+
+
+class Command(BaseCommand):
+    """
+    Links a remote domain as a downstream domain to a given upstream domain on this HQ
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('-r', '--url', required=True,
```
Comment (proteusvacuum) [inline corehq/apps/linked_domain/management/commands/add_downstream_domain.py:12]: I would call this `downstream_url` to be explicit

File: corehq/apps/linked_domain/management/commands/link_to_upstream_domain.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+class Command(BaseCommand):
+    """
+    Setup a domain on this HQ as a downstream domain to a remote upstream domain
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument('-r', '--url_base', required=True,
+                            help="Base URL of remote upstream HQ e.g. https://www.commcarehq.org")
+        parser.add_argument('-m', '--master_domain', required=True,
```
Comment (proteusvacuum) [inline corehq/apps/linked_domain/management/commands/link_to_upstream_domain.py:14]: I think we're trying to use `upstream_domain` going forwards

File: corehq/apps/linked_domain/management/commands/link_to_upstream_domain.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+                            help="Upstream master domain.")
+        parser.add_argument('-u', '--username', required=True,
+                            help="Username for remote authentication")
+        parser.add_argument('-k', '--api_key', required=True,
+                            help="ApiKey for remote authentication")
+        parser.add_argument('-d', '--domain', required=True,
```
Comment (proteusvacuum) [inline corehq/apps/linked_domain/management/commands/link_to_upstream_domain.py:20]: same here, would use `downstream_domain`

**Review by proteusvacuum — APPROVED** [review]

File: corehq/apps/linked_domain/README.md (hunk 1 above)
Comment (sravfeyn) [reply corehq/apps/linked_domain/README.md:77]: 👍

**Pull request merged**